### ISSUE TITLE
Common Types for Managed Identities

### ISSUE THREAD

**Issue by tombuildsstuff** [Azure/azure-sdk-for-go#14409]
### Feature Request

Managed Identities are of a predefined type at this point in time (although they appear to be missing [from the API contracts](https://github.com/Azure/azure-resource-manager-rpc)) - they're some combination of:

- None
- System Assigned
- System Assigned, User Assigned
- User Assigned

As such these could form a common type in `Azure/go-autorest` which would enable both code re-use for consumers of the SDK - and validation/generation errors for Swaggers which didn't match the common definitions (as seen in https://github.com/Azure/azure-rest-api-specs/issues/13340). At present these are instead uniquely generated per-type and not necessarily complete (see that Github issue for one example) - so this could be consolidated since this is a common convention/defined type.

In short: can the SDK expose common types for these, rather than a type (or multiple types) per service/struct?

Thanks!

**Comment by lirenhe** [comment]
@jhendrixMSFT , do you have any thinking for this ask?

**Comment by jhendrixMSFT** [comment]
@mikekistler is this something we can address with TypeSpec?

**Comment by mikekistler** [comment]
There is no provision for this in TypeSpec, as best I am aware.
